overlord/servicestate/service_control_test.go

```diff
@@ -0,0 +1,454 @@
+	}
+	t.Set("service-action", cmd)
+	chg.AddTask(t)
+
+	st.Unlock()
+	defer s.se.Stop()
+	c.Assert(s.o.Settle(5*time.Second), IsNil)
+	st.Lock()
+
+	c.Assert(t.Status(), Equals, state.ErrorStatus)
+	c.Check(chg.Err(), ErrorMatches, `cannot perform the following tasks:\n.*no such service: baz.*`)
+}
+
+func (s *serviceControlSuite) TestNotAService(c *C) {
+	st := s.state
+	st.Lock()
+
+	s.mockTestSnap(c)
+
+	chg := st.NewChange("service-control", "...")
+	t := st.NewTask("service-control", "...")
+	cmd := &servicestate.ServiceAction{
+		SnapName: "test-snap",
+		Action:   "start",
+		Services: []string{"someapp"},
+	}
+	t.Set("service-action", cmd)
+	chg.AddTask(t)
+
+	st.Unlock()
+	defer s.se.Stop()
+	c.Assert(s.o.Settle(5*time.Second), IsNil)
+	st.Lock()
+
+	c.Assert(t.Status(), Equals, state.ErrorStatus)
+	c.Check(chg.Err(), ErrorMatches, `cannot perform the following tasks:\n.*someapp is not a service.*`)
+}
+
+func (s *serviceControlSuite) TestStartAllServices(c *C) {
+	st := s.state
+	st.Lock()
+
+	s.mockTestSnap(c)
+
+	chg := st.NewChange("service-control", "...")
+	t := st.NewTask("service-control", "...")
+	cmd := &servicestate.ServiceAction{
+		SnapName: "test-snap",
+		Action:   "start",
+	}
+	t.Set("service-action", cmd)
+	chg.AddTask(t)
+
+	st.Unlock()
+	defer s.se.Stop()
+	c.Assert(s.o.Settle(5*time.Second), IsNil)
+	st.Lock()
+
+	c.Assert(t.Status(), Equals, state.DoneStatus)
+
+	c.Check(s.sysctlArgs, DeepEquals, [][]string{
+		{"--root", dirs.GlobalRootDir, "is-enabled", "snap.test-snap.foo.service"},
+		{"--root", dirs.GlobalRootDir, "is-enabled", "snap.test-snap.bar.service"},
+		{"start", "snap.test-snap.foo.service"},
+		{"start", "snap.test-snap.bar.service"},
+	})
+}
+
+func (s *serviceControlSuite) TestStartListedServices(c *C) {
+	st := s.state
+	st.Lock()
+
+	s.mockTestSnap(c)
+
+	chg := st.NewChange("service-control", "...")
+	t := st.NewTask("service-control", "...")
+	cmd := &servicestate.ServiceAction{
+		SnapName: "test-snap",
+		Action:   "start",
+		Services: []string{"foo"},
+	}
+	t.Set("service-action", cmd)
+	chg.AddTask(t)
+
+	st.Unlock()
+	defer s.se.Stop()
+	c.Assert(s.o.Settle(5*time.Second), IsNil)
+	st.Lock()
+
+	c.Assert(t.Status(), Equals, state.DoneStatus)
+	c.Check(s.sysctlArgs, DeepEquals, [][]string{
+		{"--root", dirs.GlobalRootDir, "is-enabled", "snap.test-snap.foo.service"},
+		{"start", "snap.test-snap.foo.service"},
+	})
+}
+
+func (s *serviceControlSuite) TestStartEnableServices(c *C) {
+	st := s.state
+	st.Lock()
+
+	s.mockTestSnap(c)
+
+	chg := st.NewChange("service-control", "...")
+	t := st.NewTask("service-control", "...")
+	cmd := &servicestate.ServiceAction{
+		SnapName:       "test-snap",
+		Action:         "start",
+		ActionModifier: "enable",
+		Services:       []string{"foo"},
+	}
+	t.Set("service-action", cmd)
+	chg.AddTask(t)
+
+	st.Unlock()
+	defer s.se.Stop()
+	c.Assert(s.o.Settle(5*time.Second), IsNil)
+	st.Lock()
+
+	c.Assert(t.Status(), Equals, state.DoneStatus)
+	// XXX
```

Comment:
what's the XXX for?

Reply:
Ah, I meant to add a comment saying that the check for "is-enabled" will go away soon (with the rest of services work) but forgot; it doesn't make sense to add any comment about that after all because there are too many places affected the same, so removed.